packages/interweave/src/Parser.js

```diff
@@ -441,6 +441,21 @@ export default class Parser {
           return;
         }
 
+        // Apply transformation if available
+        const children = this.parseNode(nextNode, config);
+        const transformed = transform ? transform(nextNode, children, config) : null;
+        if (transformed === null) {
+          return;
+        } else if (transformed !== undefined) {
+          this.keyIndex += 1;
+          content.push((
+            <React.Fragment key={this.keyIndex}>
```

Comment:
What is the fragment for?

Reply:
The fragment allows us to attach the `key` for the array, but otherwise does nothing. We could alternatively `React.cloneElement` on the result, but I'd rather avoid that.

Reply:
I'd rather not use a fragment as it would be a breaking change.

Reply:
How so? This would be opt-in behaviour, since if you don't declare `transform` you'll end up with the same tree as always, so it wouldn't break any existing behaviour.

Reply:
Oh right, upgrading React would be breaking 👍 Will update.